r/tests/testthat/test-Table.R

```diff
@@ -145,7 +145,55 @@ test_that("[, [[, $ for Table", {
   expect_data_frame(tab[0], tbl[0])
 })
 
+test_that("[[<- assignment", {
+  tbl <- tibble::tibble(
+    int = 1:10,
+    dbl = as.numeric(1:10),
+    lgl = sample(c(TRUE, FALSE, NA), 10, replace = TRUE),
+    chr = letters[1:10],
+    fct = factor(letters[1:10])
+  )
+  tab <- Table$create(tbl)
+
+  # can remove a column
+  tab[["chr"]] <- NULL
+  expect_data_frame(tab, tbl[-4])
+
+  # can add a named column
+  tab[["new"]] <- letters[10:1]
+  expect_vector(tab[["new"]], letters[10:1])
+
+  # can add a column
+  tab[[2]] <- as.numeric(10:1)
+  expect_vector(tab[[2]], as.numeric(10:1))
+
+  # can add a column by index
+  tab[[6]] <- as.numeric(10:1)
+  expect_vector(tab[[6]], as.numeric(10:1))
+  expect_vector(tab[["6"]], as.numeric(10:1))
+
+  # can replace a column
+  tab[["int"]] <- 10:1
+  expect_vector(tab[["int"]], 10:1)
+
+  # can use $
+  tab$new <- NULL
+  expect_null(as.vector(tab$new))
+  expect_identical(dim(tab), c(10L, 5L))
+
+  tab$int <- 1:10
+  expect_vector(tab$int, 1:10)
+})
```

Comment:
Other tests we might want to add (and implement as necessary):

* `tab[[NA]] <- something`
* `tab[[NULL]] <- something`
* R input recycling (making sure we're consistent with how [[<-.data.frame works)
* assigning in an Arrow Array or ChunkedArray. `chunked_array()` should handle that but we should confirm
* any others?